internal/stats/aggregate.go

```diff
@@ -0,0 +1,212 @@
+package stats
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type AggregationQuery struct {
+	EnableAggregation bool     `json:"enable-aggregation,omitempty"`
+	GroupByFields     []string `json:"group-by,omitempty"`
+	AggregateField    string   `json:"aggregate,omitempty"`
+}
+
+type QueryResponse struct {
+	Response []map[string]any `json:"response"`
+}
+
+const (
+	COUNT_KEY  = "count"
+	MAX_KEY    = "max"
+	MEAN_KEY   = "mean"
+	MEDIAN_KEY = "median"
+	MIN_KEY    = "min"
+)
+
+// Aggregate metrics based on groupByFields and the aggregateField.
+func ApplyAggregation(metrics []map[string]any, metricType any, groupByFields []string, aggregateField string) ([]map[string]any, error) {
```

Comment:
Using `metrics` and `metricType` here seems a bit strange to me.
You may be converting to JSON to early.
If you instead passed in a []any or []Metric, then you wouldn't have to have two fields here.